src/gui/editors/PianoRoll.cpp

```diff
@@ -662,6 +663,73 @@ void PianoRoll::clearGhostPattern()
 }
 
 
+void PianoRoll::glueNotes()
+{
+	if (hasValidPattern())
+	{
+		NoteVector selectedNotes = getSelectedNotes();
+		if (selectedNotes.empty())
+		{
+			TextFloat::displayMessage( tr( "Glue notes failed" ),
+					tr( "Please select notes to glue first." ),
+					embed::getIconPixmap( "glue", 24, 24 ),
+					3000 );
+			return;
+		}
+
+		// Make undo possible
+		m_pattern->addJournalCheckPoint();
+
+		// Sort notes on key and then pos.
+		std::sort(selectedNotes.begin(), selectedNotes.end(),
+			[](const Note * note, const Note * compareNote) -> bool
+			{
+				if (note->key() == compareNote->key())
+				{
+					return note->pos() < compareNote->pos();
+				}
+				return note->key() < compareNote->key();
+			});
+
+		QList<Note *> noteToRemove;
+
+		NoteVector::iterator note = selectedNotes.begin();
+		auto nextNote = note+1;
+		NoteVector::iterator end = selectedNotes.end();
+
+		while (note != end && nextNote != end)
+		{
+			// key and position match for glue
+			if ((*note)->key() == (*nextNote)->key()
+				&& (*nextNote)->pos() >= (*note)->pos()
+				&& (*nextNote)->pos() <= (*note)->pos()
+				+ (*note)->length())
+			{
+				(*note)->setLength((*note)->length() +
+						((*nextNote)->pos() + (*nextNote)->length()) -
+						((*note)->pos() + (*note)->length()));
```

Comment:
```suggestion
				(*note)->setLength((*nextNote)->endPos() - (*note)->length());
```

Reply:
This should work... right?

Reply:
Did you try it?

Reply:
No, but the logic makes sense since we want our new note to cover the distance from the start of this note to the end of the next one. In addition, endPos is defined as startPos + length in Note.cpp, so you can substitute that in to prove equivalence:

len + ((nPos + nLen) - (pos + len))
len + (nEnd - (pos + len)
len + nEnd - pos - len
nEnd - pos

Where n prefix is appended to properties of the next note.

Reply:
@Spekular I think you meant this:
```suggestion
				(*note)->setLength((*nextNote)->endPos() - (*note)->pos());
```

Reply:
Yes, definitely. Weird slip up on my part in the earlier suggested change 😬.